docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,21 @@ a && b()
 a, b()
 
 c = a, b;
+
+a() && function namedFunctionInExpressionContext () {f();}
+
+(function anIncompleteIIFE () {});
+
```

Comment:
Good call to remove bold format from Note because that is ordinary supporting information.